### PR TITLE
More precise robot footprint

### PR DESCRIPTION
**Pull request by lucasb-eyer**
Recent partial discussion in strands-project/strands_navigation#127 and older discussion in strands-project/autonomous_patrolling#70

I've been running with this since yesterday, and it did improve navigation in tight spots a bit.
Now that I can see the output of `navigation_base`, I didn't notice any missed messages as feared by @BFALacerda, but that might be thanks to offloading the chest-obstacle stuff to the side-pc.

I'm creating this PR for discussion and also for others to try.

### CODE REVIEW

File: strands_movebase/strands_movebase_params/costmap_common_params.yaml [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
 
-footprint: [[0.235, 0.31], [-0.515, 0.31], [-0.515, -0.31], [0.235, -0.31]]
+footprint: [[-0.30,-0.32], # Back side right
+            [-0.52,-0.11], # Back right
+            [-0.52, 0.11], # Back left
+            [-0.30, 0.32], # Back side left
+            [-0.05, 0.31], # Front side left
+            [ 0.05, 0.25], # Thick front left
+            [ 0.15, 0.13], # Thin front left
+            [ 0.20, 0.00], # Front
```
Comment (bfalacerda) [inline strands_movebase/strands_movebase_params/costmap_common_params.yaml:14]: shouldnt this be 0.235?

**Comment by lucasb-eyer** [comment]
No. I measured both placing a cubboard in front of the bumper and measuring the laser distance. I also tried to measure physically and checked if distances on the map seem to correspond distances IRL, and they do.

**Comment by bfalacerda** [comment]
but the rotation centre might not be exactly where the laser is. I'm awful with this transform stuff but I'm assuming that there's something that takes you from laser frame to the actual base footprint frame that's used for the costmap generation?

**Comment by nilsbore** [comment]
You can use metralabs footprint to determine the rotational centre. Image below: https://camo.githubusercontent.com/8300cdf3139ac91ea70b15230b0cef71a78e7d8e/68747470733a2f2f662e636c6f75642e6769746875622e636f6d2f6173736574732f343637303135392f313536313834382f31376435386466302d353033632d313165332d386338622d3766623635393734366563342e706e67

**Comment by nilsbore** [comment]
scitos
With the rectangular footprint (red).

**Comment by nilsbore** [comment]
Some MATLAB code if you want to play with it:

```
footprint1 = [-0.211159,-0.295;-0.0842764,-0.309861;-0.0347691,-0.307378;0.0137095,-0.297036;0.0599199,-0.2791;0.102681,-0.254027;0.140898,-0.222459;0.173596,-0.185203;0.199937,-0.143212;0.219249,-0.0975586;0.231037,-0.0494112;0.235,-3.89256e-07;0.231037,0.0494104;0.219249,0.0975578;0.199938,0.143211;0.173596,0.185202;0.140899,0.222458;0.102681,0.254026;0.0599206,0.279099;0.0137102,0.297036;-0.0347683,0.307378;-0.0842756,0.309861;-0.211159,0.295;-0.311159,0.25;-0.411159,0.185;-0.511159,0.10;-0.515,0;-0.511159,-0.10;-0.411159,-0.185;-0.311159,-0.25]';
footprint2 = [0.235, 0.31;-0.515, 0.31;-0.515, -0.31;0.235, -0.31]';
footprint3 = [-0.3000, -0.3200; -0.5200, -0.1100; -0.5200, 0.1100; -0.3000, 0.3200; 0.1000, 0.3200; 0.2500, 0.1100; 0.2500, -0.1100; 0.1000, -0.3200]';

footprint4 = [0.235, 0.31;-0.515, 0.31;-0.515, -0.31;0.235, -0.31; 0.415, -0.12; 0.415, 0.12]';
orig = [0; 0];

plot(footprint1(2, :), footprint1(1, :), 'b')
axis equal
hold on
%plot(footprint2(2, :), footprint2(1, :), 'r')
plot(orig(2), orig(1), '*b')

plot(footprint3(2, :), footprint3(1, :), 'g')

plot(footprint4(2, :), footprint4(1, :), 'r')
```

**Comment by nilsbore** [comment]
The first one is the one provided by metralabs

**Comment by bfalacerda** [comment]
yes, the metralabs footprint is the very accurate footprint commented on the file, and it has 0.235 as max, that's why i was assuming that's the correct measurement.

**Comment by lucasb-eyer** [comment]
Yeah, I used the tf transforms to see how far the laser is from the rotational centre when I did the physical measurements, I think it was somewhere around 7cm.

But the point is that in rviz, I assume the footprint is drawn centered to the correct point, and I assume the laser points are also offset according to the laser's location. Otherwise, nothing would make sense. Now I drive the robot right in front of a wall, and I can look at the laser points and the footprint, and I see that it's perfect, it even has a bit of slack.

footprint

So for all ROS purposes, I think this value is just fine.

**Comment by bfalacerda** [comment]
i'm sold

**Comment by marc-hanheide** [comment]
me, too

**Pull request merged**